### PR TITLE
Add tooling extension query support to Validation layers

### PR DESCRIPTION
**Pull request by mark-lunarg**
Add support to chassis for this extension, handle queries properly.

Fixes #1430.

### CODE REVIEW

**Comment by mark-lunarg** [comment]
Travis failures are due to an issue with the 1.1.130 version of the loader, which is being fixed outside this repository.

File: layers/generated/chassis.cpp [hunk 1]
```diff
@@ -902,6 +914,59 @@ VKAPI_ATTR VkResult VKAPI_CALL CreateBuffer(
...
+// Handle tooling queries manually as this is a request for layer information
+
+VKAPI_ATTR VkResult VKAPI_CALL GetPhysicalDeviceToolPropertiesEXT(
+    VkPhysicalDevice                            physicalDevice,
+    uint32_t*                                   pToolCount,
+    VkPhysicalDeviceToolPropertiesEXT*          pToolProperties) {
+    auto layer_data = GetLayerDataPtr(get_dispatch_key(physicalDevice), layer_data_map);
+    bool skip = false;
+
+    static VkPhysicalDeviceToolPropertiesEXT khronos_layer_tool_props = {
```
[the remark this answers is not in the document]
Comment (mark-lunarg) [reply layers/generated/chassis.cpp]: Fixed.

File: layers/generated/chassis.cpp [hunk 2]
```diff
@@ -902,6 +914,59 @@ VKAPI_ATTR VkResult VKAPI_CALL CreateBuffer(
...
+    VkPhysicalDeviceToolPropertiesEXT*          pToolProperties) {
+    auto layer_data = GetLayerDataPtr(get_dispatch_key(physicalDevice), layer_data_map);
+    bool skip = false;
+
+    static const VkPhysicalDeviceToolPropertiesEXT khronos_layer_tool_props = {
+        VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_TOOL_PROPERTIES_EXT,
+        nullptr,
+        "Khronos Validation Layer",
+        "VK_HEADER_VERSION",
+        VK_TOOL_PURPOSE_VALIDATION_BIT_EXT | VK_TOOL_PURPOSE_ADDITIONAL_FEATURES_BIT_EXT | VK_TOOL_PURPOSE_DEBUG_REPORTING_BIT_EXT | VK_TOOL_PURPOSE_DEBUG_MARKERS_BIT_EXT,
```
Comment (krOoze) [inline layers/generated/chassis.cpp]: I think it should only advertise `ADDITIONAL_FEATURES` if those are enabled?
Same for `DEBUG_REPORTING`. Otherwisely why would it be separate flag from `VALIDATION`?

**Comment by mark-lunarg** [comment]
If they are not enabled, there's no need to query them.  Validation would seem to be independent of any reporting extensions. But now that I've ready additional_features more closely, I'm less certain about that one -- I interpreted that to mean that the layers directly support extensions that an ICD might not, which we certainly do, but that might not be the intention.

File: layers/generated/chassis.cpp [hunk 3]
```diff
@@ -902,6 +914,59 @@ VKAPI_ATTR VkResult VKAPI_CALL CreateBuffer(
...
+    uint32_t*                                   pToolCount,
+    VkPhysicalDeviceToolPropertiesEXT*          pToolProperties) {
+    auto layer_data = GetLayerDataPtr(get_dispatch_key(physicalDevice), layer_data_map);
+    bool skip = false;
+
+    static const VkPhysicalDeviceToolPropertiesEXT khronos_layer_tool_props = {
+        VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_TOOL_PROPERTIES_EXT,
+        nullptr,
+        "Khronos Validation Layer",
+        "VK_HEADER_VERSION",
```
Comment (krOoze) [inline layers/generated/chassis.cpp]: Bug -> #1436

**Comment by krOoze** [comment]
@mark-lunarg I think this kinda misses the point of the extension. Instead it should list all the distinct tools in the layers: e.g. `threading`. `unique_objcets`, `core_validation`, `gpu_assisted_validation`, `best_practices`, etc.

**Comment by mark-lunarg** [comment]
@krOoze, I'm much inclined to leave this as generic as possible, and would only reluctantly consider enumerating different types or classes of validation within the layer, which are (now) arbitrary groupings in any case.  What do you suggest for 'description'?

**Comment by krOoze** [comment]
@mark-lunarg Giving the same info as `vkEnumerateDeviceLayers` kinda defeats the purpose of the new extension. We can already query which Layers are enabled. Given the extension has `layer` parameter feels like it exactly want to coerce layers to enumerate its arbitrary internal organization and which tools are currently running and which are not.

> What do you suggest for 'description'?

What about:

> "The Khronos Validation Layer reports incorrect and discouraged usage of the Vulkan API."

though maybe I think the extension authors should establish some style here (e.g. if name needs to be repeated here or not, if it should be sentence with period, etc)

**Comment by mark-lunarg** [comment]
@krOoze, yeah, there is some internal work related to exposing layer internals, so we'll need to keep it to a minimum for the time being -- seems like the most important value of the extension is the tooling info bits.  You're right, will update the description shortly to be something like RenderDoc's: "Debugging capture layer for RenderDoc".

**Pull request merged**